discovery_engine_core/tooling/bin/news_client.rs

```diff
@@ -12,133 +12,143 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-extern crate core;
-
-use anyhow::{Context, Result};
-use std::{env, time::Duration};
-use tokio::time::sleep;
-use xayn_discovery_engine_providers::{
-    gnews_client,
-    newscatcher_client,
-    CommonQueryParts,
-    Filter,
-    GnewsHeadlinesQuery,
-    GnewsNewsQuery,
-    HeadlinesQuery,
-    Market,
-};
+// use anyhow::{Context, Result};
+// use std::{env, time::Duration};
+// use tokio::time::sleep;
+// use xayn_discovery_engine_providers::{
+//     gnews,
+//     newscatcher,
+//     CommonQueryParts,
+//     Filter,
+//     GnewsHeadlinesQuery,
+//     GnewsNewsQuery,
+//     HeadlinesQuery,
+//     Market,
+// };
+
+//FIXME[TY-2810] Depending on what this tool is meant for fix it appropriately.
```

Comment:
afaik this tool was used for two purposes, 1) to experiment with the respective provider apis, and 2) to fetch some real data for the test fixtures.

if we don't need it anymore, we might just remove it, but it should be double checked with @x3ro

Reply:
Mmh, I think that having a tool that allows us to use our news client(s) from the command line would continue to be useful in the future. For example, I used this rather recently when downloading test data for elastic search. It's also a nice way of doing manual testing on large amounts of news data, to catch e.g. deserialisation problems that only happen occasionally. 

As a result, I'd "soft vote" to keep it, but also haven't taken a look at the changes, and thus at how difficult it would be to adapt it. Do you have a stronger opinion, @acrrd ?

Reply:
So it wants to fetch the input to the providers crate, not the output.

This means it

- can use the "encode query parameters" part
- but NOT the rest of the fetching (as it does things like filling in default values)

Furthermore non of this code really takes advantage by any abstractions this lead to a situations where the code is in practice more complicated then a small bash snippet doing the fetching.

I would at most replace it with a bash script doing  `for x in {1..10}; do curl .... > ...; sleep 0.1;  done` or remove it completely.